src/FormBuilder/resources/js/form-builder/src/common/currency.tsx

```diff
@@ -22,28 +45,50 @@ interface CurrencyControlProps extends CurrencyInputProps {
     help?: string;
 }
 
-const CurrencyControl = ({label, help, hideLabelFromVision, ...rest}: CurrencyControlProps) => {
+const CurrencyControl = ({label, help, hideLabelFromVision, value, onValueChange, ...rest}: CurrencyControlProps) => {
+    const [localizedValue, setLocalizedValue] = useState(value);
+
     const {currency = 'USD'} = getFormBuilderWindowData();
     // simplified implementation of useBaseControlProps()
     const uniqueId = useInstanceId(BaseControl, 'wp-components-base-control');
 
+    const updateValue = (value: string) => {
+        setLocalizedValue(value);
+        onValueChange(parseValueFromLocale(value));
+    };
+
     return (
         <BaseControl label={label} help={help} id={uniqueId} hideLabelFromVision={hideLabelFromVision}>
             <CurrencyInput
-                {...rest}
                 id={uniqueId}
+                value={localizedValue}
+                onValueChange={updateValue}
                 className={'components-text-control__input'}
                 allowDecimals={true}
                 allowNegativeValue={false}
                 maxLength={9}
                 intlConfig={{locale: window.navigator.language, currency}}
+                {...rest}
             />
         </BaseControl>
     );
 };
 
-export {
-    formatCurrencyAmount,
-    Currency,
-    CurrencyControl,
-};
+export {formatCurrencyAmount, Currency, CurrencyControl};
+
+// @ts-ignore
+window.givewp = window.givewp || {};
+// @ts-ignore
+window.givewp.form = window.givewp.form || {};
+window.givewp.form.components = window.givewp.form.components || {};
+window.givewp.form.components.CurrencyControl = CurrencyControl;
+
+declare const window: {
+    givewp?: {
+        form?: {
+            components?: {
+                CurrencyControl?: typeof CurrencyControl;
+            };
+        };
+    };
+} & Window;
```

Comment:
I could be wrong (@jonwaldstein ?), but I don't think this is really doing anything. This trick declares the window structure for this file without affecting the global. So it's really only checking that the assignments above match, but that's also inferred. Doesn't hurt, of course.

Reply:
Yeah, I also wouldn't assign components to the window like this, it gets a little messy and I think would actually keep assigning every render.  Assigning to the window should be elevated outside the component to either the app level or registrar level.

Reply:
Maybe in another context, but we replaced a few `@TS-ignore` with that approach the other day. That's the only reason I brought that here as well.

Reply:
I believe the subjects got mixed up, but I understood both points of view. Firstly, I moved the CurrencyControl component to the components folder. Then, I transferred all registrations of the window object to the index file in the components domain. Lastly, I registered the components object to the "global" window declaration under `getWindowData`.

Reply:
@pauloiankoski great, thanks!